org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/wizards/dialogfields/TreeListDialogField.java

```diff
@@ -646,7 +646,6 @@ public void removeAllElements() {
 		if (fElements.size() > 0) {
 			fElements.clear();
 			refresh();
-			dialogFieldChanged();
```

Comment:
This has to be restored, as the contract is to modify listeners on list changes.

Reply:
restored.